src/actions/lever/LeverFYActions.sol

```diff
@@ -0,0 +1,438 @@
+import {Lever20Actions} from "./Lever20Actions.sol";
+import {IFYPool,IFYToken} from "../vault/VaultFYActions.sol";
+
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+// WARNING: These functions meant to be used as a a library for a PRBProxy. Some are unsafe if you call them directly.
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+/// @title LeverFYActions
+/// @notice A set of vault actions for modifying positions collateralized by Yield pTokens
+contract LeverFYActions is Lever20Actions, ICreditFlashBorrower, IERC3156FlashBorrower {
+    using SafeERC20 for IERC20;
+
+    /// ======== Custom Errors ======== ///
+
+    error LeverFYActions__onFlashLoan_unknownSender();
+    error LeverFYActions__onFlashLoan_unknownToken();
+    error LeverFYActions__onFlashLoan_nonZeroFee();
+    error LeverFYActions__onFlashLoan_unsupportedAction();
+    error LeverFYActions__onCreditFlashLoan_unknownSender();
+    error LeverFYActions__onCreditFlashLoan_nonZeroFee();
+    error LeverFYActions__onCreditFlashLoan_unsupportedAction();
+    error LeverFYActions__solveTradeInvariant_tokenMismatch();
+    error VaultFYActions__toUint128_overflow();
+
+    /// ======== Types ======== ///
+
+    struct CollateralSwapParams {
+        // Min amount of asset out [tokenScale for buying, underlierScale for selling]
+        uint256 minAssetOut;
+        // Address of the Yield Space v2 pool
+        address yieldSpacePool;
+        // Address of the underlier (underlierToken) when buying, Address of the fyToken (token) when selling
+        address assetIn;
+        // Address of the fyToken (token) when buying, Address of underlier (underlierToken) when selling
+        address assetOut;
+    }
+
+    struct BuyCollateralAndIncreaseLeverFlashLoanData {
+        // Address of the collateral vault
+        address vault;
+        // Address of the collateral token
+        address token;
+        // Address of the owner of the position
+        address position;
+        // Amount of underliers the user provides (directly impacts the health factor) [underlierScale]
+        uint256 upfrontUnderliers;
+        // Swap config for the FIAT to underlier swap
+        SellFIATSwapParams fiatSwapParams;
+        // Swap config for the underlier to collateral token swap
+        CollateralSwapParams collateralSwapParams;
+    }
+
+    struct SellCollateralAndDecreaseLeverFlashLoanData {
+        // Address of the collateral vault
+        address vault;
+        // Address of the collateral token
+        address token;
+        // Address of the owner of the position
+        address position;
+        // Address of the account who receives the withdrawn / swapped underliers
+        address collateralizer;
+        // Amount of pTokens to withdraw and swap for underliers [tokenScale]
```

Comment:
replace every occurrence of pToken with fyToken

Reply:
updated both LeverFYActions.sol and t.sol